src/context/notificationStateSlice.ts

```diff
@@ -84,23 +84,39 @@ export const {
 export const errorToNotificationMiddleware: Middleware<{}, RootState> = () => (
   dispatch: AppDispatch,
 ) => (action: AnyAction) => {
-  const dispatchResult = dispatch(action);
-
-  // eslint-disable-next-line default-case
-  switch (action.type) {
-    case 'mapState/loadLayerData/rejected':
-    case 'serverState/loadAvailableDates/rejected':
-    case 'tableState/loadTable/rejected':
-      dispatch(
-        addNotification({
-          type: 'error',
-          message:
-            typeof action.error === 'string'
-              ? action.error
-              : action.error.message,
-        }),
-      );
+  let dispatchResult;
+  try {
+    // catch sync errors
+    // eslint-disable-next-line fp/no-mutation
+    dispatchResult = dispatch(action);
+  } catch (err) {
+    console.error(err);
+    dispatch(
+      addNotification({
+        type: 'error',
+        message: err.message || err,
+      }),
+    );
+    return dispatchResult;
```

Comment:
This isn't going to be defined if `dispatch` threw an error. We probably want to just re-throw the error here, or possibly return `undefined` explicitly - I find it a bit misleading to return `dispatchResult` though.